### PR TITLE
Use raw bytes for SRS points to save half storage 

### PR DESCRIPTION
**Pull request by jianoaix**
## Why are these changes needed?
Storing raw bytes instead of hex as ascii could reduce the storage of G1 and G2 points by 50%.

## Checks

- [x] I've made sure the lint is passing in this PR.
- [x] I've made sure the tests are passing. Note that there might be a few flaky tests, in that case, please comment that they are not relevant.
- Testing Strategy
   - [ ] Unit tests
   - [ ] Integration tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: pkg/kzg/bn254/bn254_all_test.go [hunk 1]
```diff
@@ -38,6 +38,9 @@ func TestPointG1Marshalling(t *testing.T) {
 
 	bytes := point.MarshalText()
 
+	// The point is serialized to raw bytes and it should be 32 bytes.
```
Comment (jianoaix) [inline pkg/kzg/bn254/bn254_all_test.go:41]: This was 64 bytes (the G2 below was 128 bytes) before this PR

File: pkg/encoding/utils/pointsIO.go [hunk 2]
```diff
@@ -10,11 +11,27 @@ import (
...
+	G2PointBytes = 64
+)
+
 type EncodeParams struct {
 	NumNodeE  uint64
 	ChunkLenE uint64
 }
 
+func ReadFile(reader *bufio.Reader, numBytesToRead uint64) ([]byte, error) {
+	buf := make([]byte, numBytesToRead)
```
Comment (bxue-l2) [inline pkg/encoding/utils/pointsIO.go:27]: Would it be better if adding a check between the bytes requested and byte actually read?

File: pkg/encoding/utils/pointsIO.go [hunk 3]
```diff
@@ -30,19 +47,19 @@ func ReadG1Points(filepath string, n uint64, numWorker uint64) ([]bls.G1Point, e
...
 
-	buf, _, err := g1r.ReadLine()
+	buf, err := ReadFile(g1r, n*G1PointBytes)
 	if err != nil {
 		return nil, err
 	}
 
-	if uint64(len(buf)) < 64*n {
-		log.Printf("Error. Insufficient G1 points. Only contains %v. Requesting %v\n", len(buf)/64, n)
+	if uint64(len(buf)) < G1PointBytes*n {
```
Comment (bxue-l2) [inline pkg/encoding/utils/pointsIO.go:61]: cool. it is checked here

**Review by bxue-l2 — APPROVED** [review]

File: pkg/encoding/utils/pointsIO.go (hunk 2 above)
Comment (jianoaix) [reply pkg/encoding/utils/pointsIO.go:27]: Yeah I was thinking that as well. But there is guarantee from API already: ReadFull() ensures that "On return, n == len(buf) if and only if err == nil" already: https://pkg.go.dev/io#ReadFull

File: pkg/encoding/utils/pointsIO.go (hunk 3 above)
Comment (bxue-l2) [reply pkg/encoding/utils/pointsIO.go:63]: wait not, the buf capacity is created to be equal

Comment (jianoaix) [reply pkg/encoding/utils/pointsIO.go]: Yea this will always false given the impl of `ReadDesiredBytes`. Cleaned up!

**Pull request merged**